browser/src/layer/tile/CanvasTileLayer.js

```diff
@@ -2948,10 +2948,33 @@ L.CanvasTileLayer = L.Layer.extend({
 			pos = this._twipsToLatLng({ x: pos.x, y: pos.y });
 
 		var center = this._map.project(pos);
-		center = center.subtract(this._map.getSize().divideBy(2));
-		center.x = Math.round(center.x < 0 ? 0 : center.x);
-		center.y = Math.round(center.y < 0 ? 0 : center.y);
-		this._map.fire('scrollto', {x: center.x, y: center.y});
+
+		// If x coordinate is already within visible area, we won't scroll to that direction.
+		if (app.file.viewedRectangle.cX1 <= center.x && center.x <= app.file.viewedRectangle.cX2)
+			center.x = app.file.viewedRectangle.cX1;
+		else {
+			center.x -= this._map.getSize().divideBy(2).x;
+			center.x = Math.round(center.x < 0 ? 0 : center.x);
+		}
+
+		// If y coordinate is already within visible area, we won't scroll to that direction.
+		const controlYDown = center.y + (app.file.textCursor.visible ? app.file.textCursor.rectangle.cHeight :
+			(app.calc.cellCursorVisible ? app.calc.cellCursorRectangle.cHeight : 0));
+
+		const controlYUp = center.y - (app.file.textCursor.visible ? app.file.textCursor.rectangle.cHeight :
+			(app.calc.cellCursorVisible ? app.calc.cellCursorRectangle.cHeight : 0));
+
+		if (app.file.viewedRectangle.cY1 <= controlYUp && controlYDown <= app.file.viewedRectangle.cY2)
+			center.y = app.file.viewedRectangle.cY1;
+		else {
+			center.y -= this._map.getSize().divideBy(2).y;
+			center.y = Math.round(center.y < 0 ? 0 : center.y);
+		}
+
+		const section = app.sectionContainer.getSectionWithName(L.CSections.Scroll.name);
+		if (section) {
+			section.onScrollTo({x: center.x, y: center.y});
```

Comment:
can we avoid to call it if both X and Y wasn't changed? as we already do these checks above

Reply:
Yes we can improve that way. Let me also do this while i check the issue with multiple users.